packages/engine/src/datastore/batch/context.rs

```diff
@@ -152,6 +152,8 @@ impl Batch {
             .zip_eq(column_dynamic_meta_list.par_iter())
             .try_for_each(|((column_writer, buffer), meta)| column_writer.write(buffer, meta))?;
 
+        self.metaversion.increment_batch();
```

Comment:
This might result in incrementing the context batch's batch version twice in a row (if the increment above in the same function incremented the memory and batch versions), but the engine and runners just check whether the batch version is greater than the previous one or not (not how much it's greater by), so it's fine.